ci/build_python.sh

```diff
@@ -5,6 +5,20 @@ set -euo pipefail
 
 rapids-configure-conda-channels
 
+# Setting channel priority per-repo until all RAPIDS can build using strict channel priority
+# This will be replaced when we port this recipe to `rattler-build`
+if [[ "$RAPIDS_CUDA_VERSION" == 11.* ]]; then
+  rapids-logger "Channel priority disabled for CUDA 11 builds"
+else
+  rapids-logger "Setting strict channel priority for CUDA 12 builds"
+  conda config --set channel_priority strict
+fi
+
+# Remove `rapidsai` channel for non-release builds
+if ! rapids-is-release-build; then
+  conda config --remove channels rapidsai
+fi
```

Comment:
Not terribly important, but this boilerplate seems like it could be added to a function in [gha-tools](https://github.com/rapidsai/gha-tools) to reduce the copy-paste stuff.

Reply:
Yep, strong agree -- I have a draft PR open for that here https://github.com/rapidsai/gha-tools/pull/153 but I didn't want to merge it in mid-release.
I think I'm going to hold off on any more channel priority PRs until I can change things more easily.